presto-tests/src/test/java/com/facebook/presto/tests/TestLocalQueriesDeterministic.java

```diff
@@ -0,0 +1,304 @@
+    }
+
+    @Override
+    protected void assertUpdate(@Language("SQL") String actual, @Language("SQL") String expected)
+            throws Exception
+    {
+        determinismChecker.checkPlanIsDeterministic(actual);
+    }
+
+    @Override
+    protected void assertUpdate(Session session, @Language("SQL") String actual, @Language("SQL") String expected)
+            throws Exception
+    {
+        determinismChecker.checkPlanIsDeterministic(actual);
+    }
+
+    @Override
+    protected void assertUpdate(@Language("SQL") String sql)
+    {
+        determinismChecker.checkPlanIsDeterministic(sql);
+    }
+
+    @Override
+    protected void assertUpdate(Session session, @Language("SQL") String sql)
+    {
+        determinismChecker.checkPlanIsDeterministic(sql);
+   }
+
+    @Override
+    protected void assertUpdate(@Language("SQL") String sql, long count)
+    {
+        determinismChecker.checkPlanIsDeterministic(sql);
+    }
+
+    @Override
+    protected void assertUpdate(Session session, @Language("SQL") String sql, long count)
+    {
+        determinismChecker.checkPlanIsDeterministic(sql);
+    }
+
+    @Override
+    protected void assertQueryFails(@Language("SQL") String sql, @Language("RegExp") String expectedMessageRegExp)
+    {
+        super.assertQueryFails(sql, expectedMessageRegExp);
+    }
+
+    @Override
+    protected void assertQueryFails(Session session, @Language("SQL") String sql, @Language("RegExp") String expectedMessageRegExp)
+    {
+        super.assertQueryFails(session, sql, expectedMessageRegExp);
+    }
+
+    @Override
+    protected void assertAccessAllowed(@Language("SQL") String sql, TestingAccessControlManager.TestingPrivilege... deniedPrivileges)
+            throws Exception
+    {
+    }
+
+    @Override
+    protected void assertAccessAllowed(Session session, @Language("SQL") String sql, TestingAccessControlManager.TestingPrivilege... deniedPrivileges)
+            throws Exception
+    {
+    }
+
+    @Override
+    protected void assertAccessDenied(@Language("SQL") String sql, @Language("RegExp") String exceptionsMessageRegExp, TestingAccessControlManager.TestingPrivilege... deniedPrivileges)
+            throws Exception
+    {
+    }
+
+    @Override
+    protected void assertAccessDenied(Session session, @Language("SQL") String sql, @Language("RegExp") String exceptionsMessageRegExp, TestingAccessControlManager.TestingPrivilege... deniedPrivileges)
+            throws Exception
+    {
+    }
+
+    @Override
+    protected void assertTableColumnNames(String tableName, String... columnNames)
+    {
+    }
+
+    @Override
+    protected MaterializedResult computeExpected(@Language("SQL") String sql, List<? extends Type> resultTypes)
+    {
+        determinismChecker.checkPlanIsDeterministic(sql);
+        return super.computeExpected(sql, resultTypes);
+    }
+
+    //Ignore queries that can't be used for plan determinism checking:
+
+    @Override
+    public void testDescribeInput()
```

Comment:
It would be great to classify tests (e.g: add group for tests that can be used to ensure deterministic planning). This kind of tests selection is very fragile (e.g: other will add irrelevant tests that will be executed here) and hard to maintain.